src/InfoCarrier.Core/Server/QueryDataHelper.cs

```diff
@@ -39,7 +39,6 @@ internal class QueryDataHelper
         private readonly DbContext dbContext;
         private readonly IEnumerable<IInfoCarrierValueMapper> valueMappers;
         private readonly System.Linq.Expressions.Expression linqExpression;
-        private readonly ITypeResolver typeResolver = new TypeResolver();
```

Comment:
To make the forward merges to 3.x and 5.x branches smoother, please fix QueryDataHelper the same way as InfoCarrierQueryCompiler, e.g. keep the field but just initialize it with the static instance.

Reply:
I have reintroduced the typeResolver field, you should have less work with the upstresm merge now.